internal/server/singleprocess/service_job.go

```diff
@@ -380,15 +378,14 @@ func (s *service) onDemandRunnerStartJob(
 		return nil, "", status.Errorf(codes.Aborted, "error configuring expiration: %s", err)
 	}
 
-	// Our start runner job can go anywhere.
-	// NOTE(mitchellh): For more targetted runners in multi-cluster mode,
-	// we'll probably want this to be controllable by runner profiles
-	// so that start tasks get sent to the right cluster.
-	job.TargetRunner = &pb.Ref_Runner{
-		Target: &pb.Ref_Runner_Any{
-			Any: &pb.Ref_RunnerAny{},
-		},
+	profile, err := s.state.OnDemandRunnerConfigGet(source.OndemandRunner)
```

Comment:
This call seems duplicative; we already have the value of `profile` as the `od *pb.OnDemandRunnerConfig` input to this method. Is that correct? The calls seem to be the same and use the same `source.OndemandRunner` in calling `s.state.OnDemandRunnerConfigGet()`